src/services

```diff
@@ -99,6 +99,20 @@ p({
   "client": "{\"class\":\"foam.nanos.auth.CachedAuthServiceProxy\",\"delegate\":{\"class\":\"foam.nanos.auth.ClientLoginAuthService\"}}"
 })
 
+p({
+  "class": "foam.nanos.boot.NSpec",
+  "name": "userLocator",
+  "serve": true,
+  "authenticate": true,
+  "boxClass": "foam.nanos.auth.UserLocatorServiceSkeleton",
+  "serviceScript": """
+    ul = new foam.nanos.auth.PreventDuplicateEmailLoginService(x);
```

Comment:
so in my head anything "locator" shouldn't "prevent "anything

Reply:
well perhaps need a new name.  As this service is meant to abstract all the filtering, restriction logic.

Reply:
@jlhughes  Could you please suggest one for the service name? thank you!

Reply:
@jlhughes @penzital @thepinkcacao how about RestrictToUniqueUserService.js?

Reply:
whats that? @nanopay-arthur

Reply:
alternative name to PreventDuplicateEmailLoginService. The purpose of the service is to return a single unique user with the given identifier (email or username), and it will only return a user in the case that the query returns a single user. We know we only return unique users when we login using a username (unique id per spid) or using an email for which a duplicate has not been found.

Reply:
I think "userLocator" should be designed with the thought in mind that its not necessarily used by auth services but rather locate a user in general

Reply:
not sure..

Reply:
How about UniqueUserLocatorService? 
And if needed we can also have UserLocatorService which can return all matches.